File: algorithms/0674.longest_continuous_increasing_subsequence/longest_continuous_increasing_subsequence.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        int result = 1;
        int curr_sequence = 1;
        int last = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            if (last >= nums[i]) {
                curr_sequence = 0;
            }
            curr_sequence += 1;
            result = max(result, curr_sequence);
            last = nums[i];
        }
        return result;
    }
};

File: algorithms/0674.longest_continuous_increasing_subsequence/longest_continuous_increasing_subsequence.rs
impl Solution {
    pub fn find_length_of_lcis(nums: Vec<i32>) -> i32 {
        if nums.len() <= 1 {
            return nums.len() as i32;
        }
        let mut result = 0;
        let mut count = 1;
        let mut last = nums[0];
        for &n in nums[1..].iter() {
            if last < n {
                count += 1;
            } else {
                count = 1;
            }
            last = n;
            result = result.max(count);
        }
        result
    }
}